examples/typescript-vault/main.ts

```diff
@@ -0,0 +1,36 @@
+import { Construct } from 'constructs';
+import { App, TerraformStack, TerraformOutput } from 'cdktf';
+import { VaultProvider, TransitSecretBackendKey, Mount } from './.gen/providers/vault'
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    new VaultProvider(this, 'vault', {
+      address: 'http://0.0.0.0:8200',
+      token: "YOUR_VAULT_TOKEN"
```

Comment:
Maybe we can use the default vault address and token environment variables. we can support `VAULT_ADDR` and `VAULT_TOKEN` the two environment variables.

Reply:
Yes, done b59d659

Reply:
Awesome, feel free to merge!